Hosni/Precurso/CALCULADORA/calculator-pro.js

```diff
@@ -0,0 +1,69 @@
+//Pedir valoers a calcular.
+let valores = [];
+function inputValues() {
+    let askValues 
+    do {
+        askValues = prompt("Introduzca el valor que desea calcular.");        
+            while(isNaN(askValues)) {
+                askValues = prompt("No a intoducido un número. Vuelva a intentarlo."); 
+            }
+        valores.push(parseInt(askValues));
+    } while(askValues != null);
+    if(askValues == null) {
+        valores.pop();
+    }
+    calculadora(valores)
+}
+
+function calculadora() {
+    let resultadoFinal = [];
+    
+        //------->OPERACIONS<----------//
+        
+        let resultadoSuma = ""; 
+        let resultadoResta = "";
+        let resultadoProd = "";
+        let resultadoDiv = "";
+        
+            if (valores.length === 1) {
+                let resultadoRaiz = Math.sqrt(valores);
+                console.log('Raiz de ' + valores + ' = ' + Math.round(resultadoRaiz * 1000) / 1000);
+                resultadoFinal.push('Raiz de ' + valores + ' = ' + Math.round(resultadoRaiz * 1000) / 1000);
+            }else{
+                for (i in valores) { 
```

Comment:
En el for in el "i" no tiene let o const, cuando lo haces así la variable se quedará como global(var) lo q puede suceder un error se hay otro "i"